generator/schemas/gen.go

```diff
@@ -0,0 +1,308 @@
+package schemas
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"path/filepath"
+
+	"go/ast"
+
+	"encoding/json"
+
+	"github.com/devfile/api/generator/genutils"
+	"github.com/iancoleman/strcase"
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-tools/pkg/crd"
+	"sigs.k8s.io/controller-tools/pkg/loader"
+
+	crdmarkers "sigs.k8s.io/controller-tools/pkg/crd/markers"
+	"sigs.k8s.io/controller-tools/pkg/genall"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+
+	"github.com/coreos/go-semver/semver"
+	"gomodules.xyz/orderedmap"
+)
+
+//go:generate go run sigs.k8s.io/controller-tools/cmd/helpgen generate:headerFile=../header.go.txt,year=2020 paths=.
+
+// +controllertools:marker:generateHelp:category=Devfile
+
+// GenerateJSONSchema drives whether a Json schema should be generated from this GO Struct type
+type GenerateJSONSchema struct {
+
+	// OmitCustomUnionMembers indicates that the Json schema gnerated from this type should omit Custom union members.
+	OmitCustomUnionMembers bool `marker:",optional"`
+
+	// Title indicates the content ot the Json Schema `title` attribute
+	Title string `marker:",optional"`
+}
+
+var (
+	jsonschemaVersionMarker  = markers.Must(markers.MakeDefinition("devfile:jsonschema:version", markers.DescribesPackage, ""))
+	jsonschemaGenerateMarker = markers.Must(markers.MakeDefinition("devfile:jsonschema:generate", markers.DescribesType, GenerateJSONSchema{}))
+)
+
+// +controllertools:marker:generateHelp
+
+// Generator generates Json schemas from the GO source code of the Kubernetes API
```

Comment:
So tiny comment: In go it's common practice to use upper case for abbreviation why not use it here?
```suggestion
// Generator generates JSON schemas from the GO source code of the Kubernetes API
```

Reply:
Fixed in commit https://github.com/devfile/api/pull/133/commits/38469ecd2f52343bef7a60071fc5bf4356baefac